common/scripted_effects/babylon_remove_infamy_effects.txt

```diff
@@ -0,0 +1,145 @@
+		name = population_var
+		multiply = 0.03
+	}
+	clamp_variable = {
+		name = population_var
+		min = 0
+		max = 50
+	}
+	change_variable = {
+		name = infamy_reduction_var
+		multiply = root.var:population_var
+	}
+	change_variable = {
+		name = infamy_reduction_var
+		multiply = -1
+	}
+}
+
+remove_subjugation_infamy = {
+	change_infamy = root.var:infamy_reduction_var
+	remove_variable = infamy_reduction_var
+}
+
+remove_infamy_worth_STATE_BASEINFAMY = {
+	#STATE = which state is being evaluated
+	#BASEINFAMY = amount of infamy the wargoal gives, base
+	#Conquer state wargoals give 5 base infamy, the rest gives 2
+	$STATE$ = {
+		save_scope_as = target_state
+	}
+	set_local_variable = {
+		name = infamy_reduction_var
+		value = $BASEINFAMY$
+	}
+	#Sidestep, get the population multiplier
+	set_local_variable = {
+		name = population_var
+		value = scope:target_state.state_population
+	}
+	change_local_variable = {
+		name = population_var
+		#One million
+		divide = 1000000
+	}
+	change_local_variable = {
+		name = population_var
+		multiply = 3
+	}
+	change_local_variable = {
+		name = population_var
+		add = 1
+	}
+	clamp_local_variable = {
+		name = population_var
+		min = 0
+		max = 5
+	}
+	#Multiplier cooked!
+	change_local_variable = {
+		name = infamy_reduction_var
+		multiply = local_var:population_var
+	}
+	change_local_variable = {
+		#Do a reversal on the var; from positive to negative since it will be removed..
+		name = infamy_reduction_var
+		multiply = -1
+	}
+	change_infamy = local_var:infamy_reduction_var
+}
+remove_infamy_worth_COUNTRY_BASEINFAMY = {
+	#COUNTRY = which (target)country is being evaluated
+	#BASEINFAMY = amount of infamy the wargoal gives, base
+	$COUNTRY$ = {
+		save_scope_as = target_country
+	}
+	set_local_variable = {
+		name = infamy_reduction_var
+		value = $BASEINFAMY$
+	}
+	#Sidestep, get the population multiplier
+	set_local_variable = {
+		name = population_var
+		value = scope:target_country.total_population
+	}
+	change_local_variable = {
+		name = population_var
+		#One million
+		divide = 1000000
+	}
+	change_local_variable = {
+		name = population_var
+		multiply = 3
+	}
+	change_local_variable = {
+		name = population_var
+		add = 1
+	}
+	clamp_local_variable = {
+		name = population_var
+		min = 0
+		max = 50
+	}
+	#Multiplier cooked!
+	change_local_variable = {
+		name = infamy_reduction_var
+		multiply = local_var:population_var
+	}
+	change_local_variable = {
+		#Do a reversal on the var; from positive to negative since it will be removed..
+		name = infamy_reduction_var
+		multiply = -1
+	}
+	change_infamy = local_var:infamy_reduction_var
+}
+
+babylon_reset_infamy_to_month_start = {
+	set_local_variable = {
+		name = infamy_difference
+		value = var:infamy_on_month_start
+	}
+	change_local_variable = {
+		name = infamy_difference
+		subtract = infamy
+	}
+	change_infamy = local_var:infamy_difference
+}
+
+test_effect = {
+	debug_log = "my stuff"
+	debug_log_scopes = yes
+}
```

Comment:
Whoops, looks like I left a debug_effect in! Oh well, I don't think it applies anywhere.